core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -860,6 +864,16 @@ private void updateGraphHooks(final GraphConfig config) {
                     config.getHooks().add(0, new NamedOperationResolver());
                 }
             }
+            boolean hasNamedViewHook = false;
+            for (final GraphHook graphHook : config.getHooks()) {
```

Comment:
You could merge this for loop and the one before together to avoid having to iterate over the hooks twice.